### PR TITLE
Feature/2 ecs

### PR DESCRIPTION
**Pull request by CedricCORGE**

### CODE REVIEW

File: src/ECS/Components/AccelerationComponent.hpp [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#pragma once
```
Comment (tbellicha) [inline src/ECS/Components/AccelerationComponent.hpp:1]: Missing header

File: src/ECS/Registry.hpp [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        /*if (_components_arrays.contains(std::type_index(typeid(Component))) == false) {
+            // TODO: throw
+            return sparse_array<Component>();
+        }*/
+        return std::any_cast<sparse_array<Component> &>(_components_arrays[std::type_index(typeid(Component))]);
+    }
+    template <class Component> sparse_array<Component> const &get_components() const
+    {
+        /*if (_components_arrays.contains(std::type_index(typeid(Component))) == false) {
+            // TODO: throw
```
Comment (Djadanis) [inline src/ECS/Registry.hpp:50]: Please don't put TODOs in your code !

File: src/ECS/Components/HealthComponent.hpp [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#pragma once
```
Comment (tbellicha) [inline src/ECS/Components/HealthComponent.hpp:1]: Missing header

File: src/ECS/Registry.hpp [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        return this->add_component(to, c);
+    }
+    template <typename Component> void remove_component(Entity const &from)
+    {
+        auto &arr = this->get_components<Component>();
+
+        arr.erase(from);
+    }
+
+    /*void create_square(int length, int width, int x, int y, int x_velo, int y_velo, float refresh_time, bool collide)
```
Comment (Djadanis) [inline src/ECS/Registry.hpp:103]: Comments

File: src/ECS/Components/MaximumVelocityComponent.hpp [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#pragma once
```
Comment (tbellicha) [inline src/ECS/Components/MaximumVelocityComponent.hpp:1]: Missing header

File: src/ECS/Components/PositionComponent.hpp [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#pragma once
```
Comment (tbellicha) [inline src/ECS/Components/PositionComponent.hpp:1]: Missing header

File: src/ECS/Registry.hpp [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+    {
+        /*if (_components_arrays.contains(std::type_index(typeid(Component))) == false) {
+            // TODO: throw
+            return sparse_array<Component>();
+        }*/
+        return std::any_cast<sparse_array<Component> &>(_components_arrays[std::type_index(typeid(Component))]);
+    }
+    template <class Component> sparse_array<Component> const &get_components() const
+    {
+        /*if (_components_arrays.contains(std::type_index(typeid(Component))) == false) {
```
Comment (Louismaestre) [inline src/ECS/Registry.hpp:49]: delete com

File: src/ECS/Components/ShapeComponent.hpp [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    int width;
+    std::string text;
+    std::string font;
+    int font_size;
+
+    /*ShapeComponent(int set_radius) : radius(set_radius)
+    {
+        type = shape_type::CIRCLE;
+        x_scale = 0.0;
+        y_scale = 0.0;
```
Comment (Louismaestre) [inline src/ECS/Components/ShapeComponent.hpp:38]: delete com

**Review by Djadanis — CHANGES_REQUESTED** [review]

File: src/ECS/Components/ShapeComponent.hpp [hunk 9]
```diff
@@ -0,0 +1,67 @@
+/**
```
Comment (tbellicha) [inline src/ECS/Components/ShapeComponent.hpp:1]: Delete the file

File: src/ECS/Components/VelocityComponent.hpp [hunk 10]
```diff
@@ -0,0 +1,16 @@
+#pragma once
```
Comment (tbellicha) [inline src/ECS/Components/VelocityComponent.hpp:1]: Missing header

File: src/ECS/Components/WeaponComponent.hpp [hunk 11]
```diff
@@ -0,0 +1,14 @@
+#pragma once
```
Comment (tbellicha) [inline src/ECS/Components/WeaponComponent.hpp:1]: Missing header

File: src/ECS/Entity.hpp [hunk 12]
```diff
@@ -0,0 +1,14 @@
+#pragma once
```
Comment (tbellicha) [inline src/ECS/Entity.hpp:1]: Missing header

File: src/ECS/Registry.hpp [hunk 13]
```diff
@@ -0,0 +1,188 @@
+#pragma once
```
Comment (tbellicha) [inline src/ECS/Registry.hpp:1]: Missing header

File: src/ECS/SparseArray.hpp [hunk 14]
```diff
@@ -0,0 +1,110 @@
+#pragma once
```
Comment (tbellicha) [inline src/ECS/SparseArray.hpp:1]: Missing header

File: src/ECS/Systems/ControllableSystem.cpp [hunk 15]
```diff
@@ -0,0 +1,50 @@
+#include <iostream>
```
Comment (tbellicha) [inline src/ECS/Systems/ControllableSystem.cpp:1]: Missing header

File: src/ECS/Systems/DrawableSystem.cpp [hunk 16]
```diff
@@ -0,0 +1,70 @@
+#include "../Components/AllComponents.hpp"
```
Comment (tbellicha) [inline src/ECS/Systems/DrawableSystem.cpp:1]: Missing header

File: src/ECS/Systems/VelocitySystem.cpp [hunk 17]
```diff
@@ -0,0 +1,33 @@
+#include "../Components/AllComponents.hpp"
```
Comment (tbellicha) [inline src/ECS/Systems/VelocitySystem.cpp:1]: Missing Header

**Review by tbellicha — CHANGES_REQUESTED** [review]

**Review by Djadanis — APPROVED** [review]

**Review by tbellicha — APPROVED** [review]

**Pull request merged**